src/auth/authclient.ts

```diff
@@ -317,11 +325,22 @@ export abstract class AuthClient
     ) {
       headers.set('x-goog-user-project', this.quotaProjectId);
     }
+    if (
+      !headers.has('x-goog-allowed-locations') &&
+      this.trustBoundary &&
+      this.trustBoundary.encodedLocations
```

Comment:
Checking that `encodedLocations` is not an empty string is not enough.
We only want to set the header if the value is not no-op (0x0)